src/app/services/twitterApiRequests.js

```diff
@@ -0,0 +1,64 @@
+export function getChannel(token, searchParams) {
+	return new Promise((resolve, reject) => {
+		let twitterRequest = new XMLHttpRequest();
+		twitterRequest.open('GET', `https://cors-anywhere.herokuapp.com/https://api.twitter.com/2/users/by?usernames=${searchParams}&user.fields=description,name,profile_image_url,username,public_metrics`);
+		twitterRequest.setRequestHeader('Authorization', `Bearer ${token}`);
+		twitterRequest.send();
+		twitterRequest.onload = function () {
+			if (twitterRequest.status != 200) {
+				reject({
+					code: twitterRequest.status,
+					status: twitterRequest.statusText
+				});
+			} else {
+				const channelData = JSON.parse(twitterRequest.responseText).data[0];
+				resolve(channelData);
+			}
```

Comment:
you can move it to a separate method and reuse it where in else section will be called callback function :) DRY principle